MongoLabKit/Sources/Classes/MongoLabClient.swift

```diff
@@ -48,20 +48,22 @@ extension MongoLabClient {
 
 extension MongoLabClient {
 
-    private func dataTaskWithRequest(request: NSMutableURLRequest, completion: Completion) -> NSURLSessionTask {
-        let dataTask = NSURLSession.sharedSession().dataTaskWithRequest(request) {
-            [weak self] (data: NSData?, response: NSURLResponse?, error: NSError?) in
+    fileprivate func dataTask(with request: URLRequest, completion: @escaping Completion) -> URLSessionTask {
+        let dataTask = URLSession.shared.dataTask(with: request) {
+            [weak self] data, response, error in
 
-            self?.parseData(data, response: response, error: error, completion: completion)
+            self?.parse(data, response: response, error: error, completion: completion)
+
+            return
```

Comment:
Hey, just curious but why is the return necessary?

Also just an idea but you can bind self as follows 

`if let strongSelf = self`

but optional works too of course